storage/innobase/include/srw_lock.h

```diff
@@ -282,6 +282,8 @@ class ssux_lock_impl
 #endif
   }
 
+  bool rd_u_upgrade_try() { return writer.wr_lock_try(); }
```

Comment:
May be better to reduce the S lock count (s_unlock) here if the attempt is successful. That makes the upgrade to U complete. Currently it is being done in the caller.

Reply:
I think that it could be confusing and error prone to release a lock in a `trylock` operation. However, I can write a convenience member function `sux_lock::s_x_upgrade()` that is usable for `block_lock` but not necessarily elsewhere. The reasoning would be that `buf_page_t` is typically also protected by `fix()` and a `dict_index_t::lock` or some higher-level lock while we end up waiting.